apps/examples/stack_protection/stack_protection_main.c

```diff
@@ -54,13 +54,15 @@ int stack_prot_main(int argc, char *argv[])
 			ptr = &array_normal[STACK_NORMAL_SIZE];
 			for (i = 0; i < STACK_NORMAL_SIZE; i++) {
 				*ptr = 'a';
+				printf(" sangam ptr -----0x%x\n", ptr);
```

Comment:
I think this is not the right way to test stack protection with stack limit register. 
The idea behind stack limit register is that an exception will be triggered if the value of SP register goes below the value of limit register. The sp value will change only based on the variables / array we declare in the api.. or due to the function recursion. In our case, we are defining a array of normal size (10 x 4 bytes) which is well within the stack limit. So, no exception will be triggered in any case when we run this program. 
On the other hand, if you define a array of OVERFLOW size, then the stack limit exception will be triggered as soon as this function is called, even without user selecting any option. This is because, if we create an array of 1000 integers, then compiler generates instructions like below:

00000000 <main>:
   0:	e92d4800 	push	{fp, lr}
   4:	e28db004 	add	fp, sp, #4
   8:	e24ddefa 	sub	sp, sp, #4000	; 0xfa0

At this line stack limit exception is triggered.

Reply:
Yes, I am able to get usage fault after the limit value. Modified the test case for it